channeldb/invoices.go

```diff
@@ -2186,6 +2147,84 @@ func (d *DB) updateInvoice(hash *lntypes.Hash, refSetID *invpkg.SetID, invoices,
 	return &invoice, nil
 }
 
+// cancelHTLCs tries to cancel the htlcs in the given InvoiceUpdateDesc.
+//
+// NOTE: cancelHTLCs updates will only use the `CancelHtlcs` field in the
+// InvoiceUpdateDesc.
```

Comment:
Good catch! It is true that before the code had support to add and cancel htlcs in the same update operation (even when it was never  used). I deleted the use of `AddHtlcs` in this function :+1:

Reply:
Since it's a behavior change rather than pure refactoring, could you add some commit messages to explain why it's ok to remove this check?
```go
// Consistency check to verify that there is no overlap between
// the add and cancel sets.
if _, added := update.AddHtlcs[key]; added {
	return nil, fmt.Errorf("added htlc %v canceled", key)
}
```